packages/storage/src/append_store.rs

```diff
@@ -162,16 +226,42 @@ impl<'a, T: Serialize + DeserializeOwned, Ser: Serde> AppendStore<'a, T, Ser> {
         let len = self.get_len(storage)?;
 
         if pos >= len {
-            return Err(StdError::generic_err("DequeStorage access out of bounds"));
+            return Err(StdError::generic_err("AppendStore access out of bounds"));
         }
-        let item = self.get_at_unchecked(storage, pos);
-
-        for i in pos..(len - 1) {
-            let element_to_shift = self.get_at_unchecked(storage, i + 1)?;
-            self.set_at_unchecked(storage, i, &element_to_shift)?;
+        let max_pos = len - 1;
+        let max_page = self._page_from_position(max_pos);
+        let pos_page = self._page_from_position(pos);
+
+        match pos_page.cmp(&max_page) {
+            std::cmp::Ordering::Less => {
+                // shift items from indexes to indexes
+                let mut past_indexes: Vec<Vec<u8>> = self._get_indexes(storage, pos_page)?;
```

Comment:
Again naming, not super important but I think would be more intuitive to call it `past_values`, `values`, etc..